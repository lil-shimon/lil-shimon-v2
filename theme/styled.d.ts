import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    mediaQuery: {
      sp: string;
      tablet: string;
      laptop: string;
    };
  }
}
