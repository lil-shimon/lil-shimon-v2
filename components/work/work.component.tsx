import * as Styles from "./work.styles";

export const Work = () => {
  return (
    <Styles.Container>
      <Styles.Title>Work</Styles.Title>
      <Styles.Text>
        新卒でエンジニアになり、2年間正社員エンジニアを経験しました。その後はフリーランスとして活動しています。
        正社員の時はWebフロントエンド、バックエンド、インフラ、Windowsアプリ、モバイルアプリ、スマートウォッチアプリ等、領域問わず開発に携わりました。
        その時にプロジェクトマネージャや開発責任者の役割も経験しています。
        独立してからは主にWebフロントエンドに特化しています。直近ではNext.jsをメインで使っています。
      </Styles.Text>
    </Styles.Container>
  );
};
