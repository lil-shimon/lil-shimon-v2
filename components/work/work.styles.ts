import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  gap: 1.75rem;
  flex-direction: column;
`;

export const Title = styled.h1`
  padding-bottom: 0.5em;
  border-bottom: #333 1px dotted;
`;

export const Text = styled.p`
  text-indent: 1em;
`;
