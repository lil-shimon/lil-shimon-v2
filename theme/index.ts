import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

export const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};
