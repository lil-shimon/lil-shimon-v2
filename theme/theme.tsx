import { ReactNode } from "react";
import { ThemeProvider as StyledComponentThemeProvider } from "styled-components";

import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};

type Props = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
  return (
    <StyledComponentThemeProvider theme={theme}>
      {children}
    </StyledComponentThemeProvider>
  );
};
