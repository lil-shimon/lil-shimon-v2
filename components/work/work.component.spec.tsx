import { ThemeProvider } from "@/theme";
import { Work } from "./work.component";
import { render, screen } from "@testing-library/react";
import { expect } from "vitest";

describe("Work", () => {
  it("should render", () => {
    const { asFragment } = render(<Work />, {
      wrapper: ({ children }) => <ThemeProvider>{children}</ThemeProvider>,
    });

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("Work");
    expect(
      screen.getByText(
        "新卒でエンジニアになり、2年間正社員エンジニアを経験しました。その後はフリーランスとして活動しています。 正社員の時はWebフロントエンド、バックエンド、インフラ、Windowsアプリ、モバイルアプリ、スマートウォッチアプリ等、領域問わず開発に携わりました。 その時にプロジェクトマネージャや開発責任者の役割も経験しています。 独立してからは主にWebフロントエンドに特化しています。直近ではNext.jsをメインで使っています。"
      )
    );
  });
});
