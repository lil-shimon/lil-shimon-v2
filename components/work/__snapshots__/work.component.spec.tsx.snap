// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Work > should render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb jAMbyg"
  >
    <h1
      class="sc-gtLWhw gImffP"
    >
      Work
    </h1>
    <p
      class="sc-egkSDF byTfEP"
    >
      新卒でエンジニアになり、2年間正社員エンジニアを経験しました。その後はフリーランスとして活動しています。 正社員の時はWebフロントエンド、バックエンド、インフラ、Windowsアプリ、モバイルアプリ、スマートウォッチアプリ等、領域問わず開発に携わりました。 その時にプロジェクトマネージャや開発責任者の役割も経験しています。 独立してからは主にWebフロントエンドに特化しています。直近ではNext.jsをメインで使っています。
    </p>
  </div>
</DocumentFragment>
`;
